static/app/views/dashboardsV2/gridLayout/dashboard.tsx

```diff
@@ -0,0 +1,95 @@
+import 'react-grid-layout/css/styles.css';
+import 'react-resizable/css/styles.css';
+
+import React from 'react';
+import RGL, {WidthProvider} from 'react-grid-layout';
+
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+
+import AddWidget, {ADD_WIDGET_BUTTON_DRAG_ID} from '../addWidget';
+import {Dashboard as DnDKitDashboard} from '../dashboard';
+import {DisplayType, MAX_WIDGETS, Widget} from '../types';
+
+import SortableWidget from './sortableWidget';
+import {DRAG_HANDLE_ID} from './widgetCard';
+
+const NUM_COLS = 6;
+const ROW_HEIGHT = 120;
+const WIDGET_MARGINS = [16, 16];
+const ADD_BUTTON_POSITION = {
+  x: 0,
+  y: Number.MAX_VALUE,
+  w: 2,
+  h: 1,
+  isResizable: false,
+};
+const DEFAULT_WIDGET_WIDTH = 2;
+
+const GridLayout = WidthProvider(RGL);
+
+class Dashboard extends DnDKitDashboard {
+  renderWidget(widget: Widget, index: number) {
+    const {isEditing} = this.props;
+
+    const key = generateWidgetId(widget, index);
+    const dragId = key;
+
+    const layout = {
+      x: (DEFAULT_WIDGET_WIDTH * index) % NUM_COLS,
+      y: Number.MAX_VALUE,
+      w: DEFAULT_WIDGET_WIDTH,
+      h: widget.displayType === DisplayType.BIG_NUMBER ? 1 : 2,
+      minH: widget.displayType === DisplayType.BIG_NUMBER ? 1 : 2,
+    };
+
+    return (
+      <div key={key} data-grid={layout}>
```

Comment:
We can get around having this wrapper div by forwarding refs to the `SortableWidget` and `AddWidget` but I decided to keep this for now because it's simpler and it means I don't have to do any overriding or copying those files to support the ref

`SortableWidget` has props like `dragId` that we won't need in RGL world, so I think I can address this refs thing when we get closer to the point of totally removing dndkit from dashboards

Reply:
https://github.com/react-grid-layout/react-grid-layout#custom-child-components-and-draggable-handles

Relevant link.